### PR TITLE
Add missing default config options in plugin's reference.conf

### PR DESCRIPTION
**Pull request by migesok**
- without it, if clients tried to partially override an option inside a nested section, it failed at runtime:
```
pureconfig.error.ConfigReaderException: Cannot convert configuration to a scala.runtime.Nothing$. Failures are:
  at 'data-integrity':
    - (deployments/common/persistence.conf @ file:/Users/msokolov/prog/evo/rbow/betting/target/scala-2.13/classes/deployments/common/persistence.conf: 318) Key not found: 'correlate-events-with-meta'.
```
- added docs to config options
- fixed ConsumerPool max size potentially going below 1

### CODE REVIEW

File: persistence/src/main/resources/reference.conf [hunk 1]
```diff
@@ -53,6 +53,26 @@ evolutiongaming.kafka-journal.persistence {
     }
 
     json-codec = default
+
+    # Configuration for the dynamic pool of Kafka consumers used on recovery in case the data is not yet replicated to
+    # Cassandra
+    consumer-pool {
+      # max pool size = multiplier x number of cores, capped at 1 below
```
Comment (mr-git) [inline persistence/src/main/resources/reference.conf:60]:
```suggestion
      # max pool size = multiplier x number of cores, minimal 1
```

File: journal/src/main/scala/com/evolutiongaming/kafka/journal/ConsumerPool.scala [hunk 2]
```diff
@@ -26,10 +26,8 @@ private[journal] object ConsumerPool {
     for {
       cores <- Runtime[F].availableCores.toResource
       pool <- consumer.toResourcePool(
-        (cores.toDouble * poolConfig.multiplier)
-          .round
-          .toInt,
-        poolConfig.idleTimeout,
+        maxSize               = math.max(1, (cores.toDouble * poolConfig.multiplier).intValue),
```
[the remark this answers is not in the document]
Comment (migesok) [reply journal/src/main/scala/com/evolutiongaming/kafka/journal/ConsumerPool.scala:29]: I apply `max(1, ` to get 1 as a minimum

File: journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala [hunk 3]
```diff
@@ -325,6 +325,13 @@ object Journal {
     implicit val configReaderCallTimeThresholds: ConfigReader[CallTimeThresholds] = deriveReader[CallTimeThresholds]
   }
 
+  /**
+   * Configuration for the dynamic pool of Kafka consumers used on recovery in case the data is not yet replicated to
+   * Cassandra
+   *
+   * @param multiplier max pool size = multiplier x number of cores, capped at 1 below
```
[the remark this answers is not in the document]
Comment (migesok) [reply journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala:332]: Not sure if "minimal 1" is correct either - minimal is an adjective, used in phrases like "minimal length". More like "minimum (is) 1)". Let me rephrase it better.

[the remark this answers is not in the document]
Comment (migesok) [reply journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala]: @mr-git should be better now, please check it out

**Review by mr-git — APPROVED** [review]
